core/src/main/java/com/rudderstack/android/sdk/core/DBPersistentManager.java

```diff
@@ -480,24 +470,48 @@ private boolean checkIfStatusColumnExists(SQLiteDatabase database) {
                         }
                     } while (allRows.moveToNext());
                 }
-                allRows.close();
             } else {
                 RudderLogger.logError("DBPersistentManager: checkIfStatusColumnExists: database is not readable, hence we cannot check the existence of status column");
             }
         } catch (SQLiteDatabaseCorruptException ex) {
             RudderLogger.logError("DBPersistentManager: checkIfStatusColumnExists: Exception while checking the presence of status column due to " + ex.getLocalizedMessage());
+        } finally {
+            if (allRows != null) {
+                allRows.close();
+            }
         }
         return false;
     }
 
     void checkForMigrations() {
-        SQLiteDatabase database = getWritableDatabase();
-        if (!checkIfStatusColumnExists(database)) {
-            RudderLogger.logDebug("DBPersistentManager: checkForMigrations: Status column doesn't exist in the events table, hence performing the migration now");
-            performMigration(database);
-            return;
+        Runnable runnable = () -> {
+            try {
+                SQLiteDatabase database = getWritableDatabase();
+                if (!checkIfStatusColumnExists(database)) {
+                    RudderLogger.logDebug("DBPersistentManager: checkForMigrations: Status column doesn't exist in the events table, hence performing the migration now");
+                    performMigration(database);
+                    return;
+                }
+                RudderLogger.logDebug("DBPersistentManager: checkForMigrations: Status column exists in the table already, hence no migration required");
+            } catch (SQLiteDatabaseCorruptException ex) {
+                RudderLogger.logError(DBPERSISTENT_MANAGER_CHECK_FOR_MIGRATIONS_TAG + ex.getLocalizedMessage());
+            } catch (ConcurrentModificationException ex) {
+                RudderLogger.logError(DBPERSISTENT_MANAGER_CHECK_FOR_MIGRATIONS_TAG + ex.getLocalizedMessage());
+            } catch (NullPointerException ex) {
+                RudderLogger.logError(DBPERSISTENT_MANAGER_CHECK_FOR_MIGRATIONS_TAG + ex.getLocalizedMessage());
+            }
+        };
+        // Need to perform db operations on a separate thread to support strict mode.
+        Future<?> future = Executors.newSingleThreadExecutor().submit(runnable);
```

Comment:
There is no point in creating single thread executors at method level, we can have one single thread executor at class level, and use it for all operations.

Reply:
yes, i agree with that, will change it

Reply:
addressed it